Inspector/validate.c

```diff
@@ -0,0 +1,32 @@
+#include "validate.h"
+
+#include <CommonCrypto/CommonCrypto.h>
+
+
+
+bool validate_sha1(const void *data, size_t length, size_t slot, const CS_CodeDirectory *cd) {
+    uint8_t digest[CC_SHA1_DIGEST_LENGTH + 1] = {0, };
+
+    CC_SHA1(data, (CC_LONG)length, digest);
+    void *hash = (void *)((char *)cd + ntohl(cd->hashOffset) + cd->hashSize * slot);
+    return (memcmp(digest, hash, 20) == 0);
+}
+
+bool validate_sha256(const void *data, size_t length, size_t slot, const CS_CodeDirectory *cd) {
+    uint8_t digest[CC_SHA256_DIGEST_LENGTH + 1] = {0, };
+
+    CC_SHA256(data, (CC_LONG)length, digest);
+    void *hash = (void *)((char *)cd + ntohl(cd->hashOffset) + cd->hashSize * slot);
+    return (memcmp(digest, hash, 20) == 0);
```

Comment:
20? That's probably not right.

(I believe I've seen duplicate of this code in one of the other files. Maybe "is_valid_sha1(const void* data, size_t length, const uint8_t* expected_sha1)" should be extracted as a utility function?)